examples/kafka-plugin/manual-partitioner/main.go

```diff
@@ -35,7 +35,7 @@ type ExamplePlugin struct {
 	subscription        chan (messaging.ProtoMessage)
 	kafkaSyncPublisher  messaging.ProtoPublisher
 	kafkaAsyncPublisher messaging.ProtoPublisher
-	kafkaWatcher        messaging.ProtoWatcher
+	kafkaWatcher        messaging.ProtoManualWatcher
```

Comment:
What about calling it ProtoPartitionWatcher since we create new one using NewPartitionWatcher().

I think that explicit mark/commit offset should be allowed independently from manual/hash (probably using New*Watcher(....,/*options*/WithAutoMarkCommit())